src/container/srv_target.c

```diff
@@ -1436,17 +1436,18 @@ cont_child_aggregate(struct ds_cont_child *cont)
 	if (rc)
 		return rc;
 
-	interval = (uint64_t)DAOS_AGG_THRESHOLD * NSEC_PER_SEC;
-	D_ASSERT(hlc > (interval * 2));
+	D_ASSERT(hlc > (*interval * 2));
 	/*
 	 * Assume 'current hlc - interval' as the highest stable view (all
 	 * transactions under this epoch is either committed or aborted).
 	 */
-	epoch_max = hlc - interval;
+	epoch_max = hlc - *interval;
 
 	/* Throttle the aggregation a bit */
-	if (cinfo.ci_hae > epoch_max - interval)
+	if (cinfo.ci_hae > epoch_max - *interval) {
+		*interval = (cinfo.ci_hae - (epoch_max - *interval)); 
```

Comment:
we should only return a non-zero 'timeout' here, in other cases, a zero 'timeout' will be returned. pleases don't mess this 'timeout' with 'interval'.

Reply:
Hmm, if the timeout is zero, then the caller does not need call schedule_timeout(), only yield.